pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -77,16 +76,18 @@ func (m *MigCluster) GetServiceAccountSecret(client k8sclient.Client) (*kapi.Sec
 
 // GetClient get a local or remote client using a MigCluster and an existing client
 func (m *MigCluster) GetClient(c k8sclient.Client) (k8sclient.Client, error) {
+	// TODO: re-enable cache after issue with caching secrets is resolved
+
 	// If MigCluster isHostCluster, reuse client
-	if m.Spec.IsHostCluster {
-		return c, nil
-	}
+	// if m.Spec.IsHostCluster {
+	// 	return c, nil
+	// }
```

Comment:
I did not comment out the `IsHostCluster` check in my test and wonder if leaving it in would help determine if this affects local clients, as well as, remote.

Reply:
Yes, there are two different things going on. Local vs. Remote clients, and OCP3 vs. OCP4 clients. In my case, local/OCP3 didn't have a caching problem, but remote/OCP4 did.